akka-stream/src/main/scala/akka/stream/impl/ActorMaterializerImpl.scala

```diff
@@ -299,8 +299,6 @@ private[akka] object ActorProcessorFactory {
     val settings = materializer.effectiveSettings(att)
     def interp(s: Stage[_, _]): (Props, Unit) = (ActorInterpreter.props(settings, List(s), materializer, att), ())
     op match {
-      case Identity(_)                ⇒ throw new AssertionError("Identity cannot end up in ActorProcessorFactory")
```

Comment:
Checking first, for every materialization, a stage that shouldn't be there at all seems a bit wasteful :)